crates/sui-types/src/messages.rs

```diff
@@ -531,6 +532,11 @@ pub struct SenderSignedData {
 }
 
 impl<S> TransactionEnvelope<S> {
+    pub fn size_for_gas_metering(&self) -> u64 {
+        // u64 is fine for TX size.
+        size_of::<Self>() as u64
```

Comment:
What's the rationale behind using static struct size as tx size? e.g. a transaction could contain a large vector of inputs, and would still lead to the same size as other transactions.
Also I wonder `size_of::<Self>() as u64` is guaranteed to be the same across machines.

Reply:
Yes there's a bug here. I should also not count `auth_sign_info` as the size is not deterministic. Will find a better way.